File: src/yanzm/example/adaptersamples/ExtendsArrayAdapterSample2.java
package yanzm.example.adaptersamples;

import java.util.ArrayList;
import java.util.List;

import android.app.ListActivity;
import android.content.Context;
import android.graphics.Color;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

public class ExtendsArrayAdapterSample2 extends ListActivity {

	public void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);

		List<bindData> data = new ArrayList<bindData>();
		data.add(new bindData("Clock", R.drawable.alermclock));
		data.add(new bindData("-", 0));
		data.add(new bindData("Camera", R.drawable.camera));
		data.add(new bindData("-", 0));
		data.add(new bindData("Book", R.drawable.bookopen));
		data.add(new bindData("-", 0));
		data.add(new bindData("Calculator", R.drawable.calculator));
		data.add(new bindData("-", 0));

		MyAdapter adapter = new MyAdapter(this, R.layout.list_item5, data);
		setListAdapter(adapter);
	}

	class bindData {
		String text;
		int imageResourceId;

		public bindData(String text, int imageResourceId) {
			this.text = text;
			this.imageResourceId = imageResourceId;
		}
	}

	static class ViewHolder {
		TextView textView;
		ImageView imageView;
	}

	public class MyAdapter extends ArrayAdapter<bindData> {
		private LayoutInflater inflater;
		private int layoutId;

		public MyAdapter(Context context, int layoutId, List<bindData> objects) {
			super(context, 0, objects);
			this.inflater = (LayoutInflater) context
					.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
			this.layoutId = layoutId;
		}

		@Override
		public View getView(int position, View convertView, ViewGroup parent) {
			ViewHolder holder;

			if (convertView == null) {
				convertView = inflater.inflate(layoutId, parent, false);
				holder = new ViewHolder();
				holder.textView = (TextView) convertView
						.findViewById(R.id.textview);
				holder.imageView = (ImageView) convertView
						.findViewById(R.id.imageview);
				convertView.setTag(holder);
			} else {
				holder = (ViewHolder) convertView.getTag();
			}
			
			if(!isEnabled(position))
				convertView.setBackgroundColor(Color.BLACK);
			else {
				bindData data = getItem(position);
				holder.textView.setText(data.text);
				holder.imageView.setImageResource(data.imageResourceId);				
			}
			
			return convertView;
		}
		
        @Override
        public boolean isEnabled(int position) {
            return !(getItem(position).text.startsWith("-"));
        }
	}
}

File: app/src/main/java/yanzm/example/adaptersamples/ExtendsArrayAdapterSample2.kt
package yanzm.example.adaptersamples

import android.content.Context
import android.graphics.Color
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.ListView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class ExtendsArrayAdapterSample2 : AppCompatActivity() {

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val listView = ListView(this)
        setContentView(listView)

        val data = listOf(
            Item("Clock", R.drawable.ic_alarm),
            Item("-", 0),
            Item("Camera", R.drawable.ic_camera),
            Item("-", 0),
            Item("Book", R.drawable.ic_book),
            Item("-", 0),
            Item("Calculator", R.drawable.ic_calculator),
            Item("-", 0),
        )

        listView.adapter = MyAdapter(this, R.layout.list_item5, data)
    }

    data class Item(val text: String, val imageResourceId: Int)

    data class ViewHolder(val itemView: View) {
        private val textView: TextView = itemView.findViewById(R.id.textview)
        private val imageView: ImageView = itemView.findViewById(R.id.imageview)

        fun bind(item: Item) {
            textView.text = item.text
            imageView.setImageResource(item.imageResourceId)
        }
    }

    class MyAdapter(context: Context, private val layoutId: Int, objects: List<Item?>?) :
        ArrayAdapter<Item?>(context, 0, objects!!) {

        private val inflater = LayoutInflater.from(context)

        override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
            val holder: ViewHolder = if (convertView == null) {
                val view = inflater.inflate(layoutId, parent, false)
                ViewHolder(view).apply {
                    view.tag = this
                }
            } else {
                convertView.tag as ViewHolder
            }

            if (!isEnabled(position)) {
                holder.itemView.setBackgroundColor(Color.BLACK)
            } else {
                val item = getItem(position)!!
                holder.bind(item)
            }

            return holder.itemView
        }

        override fun isEnabled(position: Int): Boolean {
            return !getItem(position)!!.text.startsWith("-")
        }
    }
}
